# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fetch/InternalRequest.cpp
# Commit: 98a751f6f41d
# Full Hash: 98a751f6f41d4e8d23f38aa0aa1d483f83cd0bda
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2023-01-19 04:46:07
# Regressor Bug: 1247687
# File Overlap Count: 1
# Description:
#   Bug 1247687 - Implement csp for Module Workers; r=evilpie,asuth,rpl,ckerschb
#   
#   Depends on D155691
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D156102
# ==============================================================================

diff -r 78addb6af24d -r 98a751f6f41d dom/fetch/InternalRequest.cpp
--- a/dom/fetch/InternalRequest.cpp	Wed Jan 18 13:46:32 2023 +0000
+++ b/dom/fetch/InternalRequest.cpp	Wed Jan 18 13:46:33 2023 +0000
@@ -274,6 +274,7 @@
     case nsIContentPolicy::TYPE_SCRIPT:
       return RequestDestination::Script;
     case nsIContentPolicy::TYPE_INTERNAL_WORKER:
+    case nsIContentPolicy::TYPE_INTERNAL_WORKER_STATIC_MODULE:
       return RequestDestination::Worker;
     case nsIContentPolicy::TYPE_INTERNAL_SHARED_WORKER:
       return RequestDestination::Sharedworker;
@@ -350,6 +351,7 @@
     case nsIContentPolicy::TYPE_WEB_IDENTITY:
       return RequestDestination::_empty;
     case nsIContentPolicy::TYPE_INVALID:
+    case nsIContentPolicy::TYPE_END:
       break;
       // Do not add default: so that compilers can catch the missing case.
   }